pymeasure/instruments/attocube/adapters.py

```diff
@@ -0,0 +1,104 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import time
+
+from pymeasure.adapters import TelnetAdapter
+
+
+class AttocubeConsoleAdapter(TelnetAdapter):
+    """ Adapter class for connecting to the Attocube Standard Console. This
+    console is a Telnet prompt with password authentication.
+
+    :param host: host address of the instrument
+    :param port: TCPIP port
+    :param passwd: password required to open the connection
+    :param kwargs: Any valid key-word argument for TelnetAdapter
+    """
+    def __init__(self, host, port, passwd, **kwargs):
+        self.terminator = '\r\n'
+        super(AttocubeConsoleAdapter, self).__init__(host, port, **kwargs)
+        # clear the initial messages from the controller
+        time.sleep(self.query_delay)
+        super(AttocubeConsoleAdapter, self).read()
+        # send password and check authorization
+        self.write(passwd, checkAck=False)
+        time.sleep(self.query_delay)
+        ret = super(AttocubeConsoleAdapter, self).read()
+        authmsg = ret.split(self.terminator)[1]
+        if authmsg != 'Authorization success':
+            raise Exception("Attocube authorization failed ('%s')" % authmsg)
+        # switch console echo off
+        self.ask('echo off')
+
+    def check_acknowledgement(self, reply, msg=""):
```

Comment:
This is usually implemented on the instrument level by writing a [`check_errors`](https://pymeasure.readthedocs.io/en/latest/api/instruments/instruments.html#pymeasure.instruments.Instrument.check_errors) method, so I suggest moving it into the ANC300 class.

Reply:
Ok, I missed this option of properties. will convert to check_errors.